pkg/daemon/file_writers.go

```diff
@@ -45,12 +46,50 @@ func noOrigFileStampName(fpath string) string {
 	return filepath.Join(noOrigParentDir(), fpath+".mcdnoorig")
 }
 
-func createOrigFile(fromPath, fpath string) error {
-	if _, err := os.Stat(noOrigFileStampName(fpath)); err == nil {
-		// we already created the no orig file for this default file
-		return nil
+func createOrigFile(fromPath, fpath string, isCoreOSVariant, forceCreate bool) error {
+
+	// Declare a boolean variable - orig - to indicate whether an orig file should be created
+	// for the given path
+	// true - orig file
+	// false - noOrig file
+	var orig bool
+
+	// https://issues.redhat.com/browse/OCPBUGS-11437
+	// MCO keeps the pull secret to .orig file once it replaced
+	// Adapt a check used in function deleteStaleData for backwards compatibility: basically if the file doesn't
+	// exist in /usr/etc (on FCOS/RHCOS) and no rpm is claiming it, we assume that the orig file came from a
+	// wrongful backup of a MachineConfig file instead of a file originally on disk.
+	if _, err := exec.Command("rpm", "-qf", fpath).CombinedOutput(); err == nil {
+		// File is owned by an rpm
+		orig = true
+	} else if strings.HasPrefix(fpath, "/etc") && isCoreOSVariant {
```

Comment:
Hmm, could you remind me why we are explicitly doing this `isCoreOSVariant` check?

Reply:
The check isCoreVariant checks whether the underlying OS is FCOS or a derivative (including SCOS and RHCOS) that uses ostree + Ignition. These OSs keep the defaults in the /usr/etc directory. As a result, we add the check here to ensure that the file is in the default-keeping dir of the corresponding OS, so it should be orig-ed.

Reply:
Hmm, ok, I think I am fine either way, but to keep consistent with existing behaviour, maybe it would be best to remove the check. The path is not entirely correct regardless, and I think keeping CoreOS and non-CoreOS nodes consistent may make it easier in the future. What do you think?